File: LAB 7/5.m
clc; clear;

P = [1; 1; 1; 1];

% Scaling
S = [2 0 0 0;
     0 2 0 0;
     0 0 2 0;
     0 0 0 1];

% Rotation (Z-axis)
theta = pi/4;
R = [cos(theta) -sin(theta) 0 0;
     sin(theta)  cos(theta) 0 0;
     0           0          1 0;
     0           0          0 1];

% Translation
T = [1 0 0 3;
     0 1 0 4;
     0 0 1 5;
     0 0 0 1];

% Composite matrix
C = T * R * S;

P_new = C * P;

disp('Composite Transformed Point:');
disp(P_new);

File: LAB 7/5.py
import numpy as np

P = np.array([1, 1, 1, 1])

S = np.array([
    [2, 0, 0, 0],
    [0, 2, 0, 0],
    [0, 0, 2, 0],
    [0, 0, 0, 1]
])

theta = np.pi / 4
R = np.array([
    [np.cos(theta), -np.sin(theta), 0, 0],
    [np.sin(theta),  np.cos(theta), 0, 0],
    [0,              0,             1, 0],
    [0,              0,             0, 1]
])

T = np.array([
    [1, 0, 0, 3],
    [0, 1, 0, 4],
    [0, 0, 1, 5],
    [0, 0, 0, 1]
])

C = T @ R @ S
P_new = C @ P

print("Composite Transformed Point:", P_new)